src/Backups/BackupUtils.h

```diff
@@ -20,4 +21,8 @@ AccessRightsElements getRequiredAccessToBackup(const ASTBackupQuery::Elements &
 bool compareRestoredTableDef(const IAST & restored_table_create_query, const IAST & create_query_from_backup, const ContextPtr & global_context);
 bool compareRestoredDatabaseDef(const IAST & restored_database_create_query, const IAST & create_query_from_backup, const ContextPtr & global_context);
 
+/// Returns true if this table should be skipped while making a backup because it's an inner table.
+bool isInnerTableShouldBeSkippedForBackup(const QualifiedTableName & table_name);
+bool isInnerTableShouldBeSkippedForBackup(const String & database_name, const String & table_name);
```

Comment:
Maybe we can call these functions `isInnerTable` as we already have a bool to check if it needs to be skipped `skip_if_inner_table` . what do you think ?